src/List.mo

```diff
@@ -14,9 +22,27 @@ module {
   public type List<T> = ?(T, List<T>);
 
   /// Create an empty list.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// List.nil<Nat>() // => null
+  /// ```
+  ///
+  /// Runtime: O(1)
+  ///
+  /// Space: O(1)
   public func nil<T>() : List<T> = null;
 
   /// Check whether a list is empty and return true if the list is empty.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// List.isNil<Nat>(null) // => true
+  /// ```
+  ///
+  /// Runtime: O(1)
+  ///
+  /// Space: O(1)
   public func isNil<T>(l : List<T>) : Bool {
```

Comment:
```suggestion
  public func isNil<T>(list : List<T>) : Bool {
```
Seems easier to read to use whole words. What do you think?

Reply:
Except that our own style guidelines say this:
-   Use longer, more descriptive names for global or public identifier or ones with large scope, and short names for local ones with small scope.

    It is fine to use single character identifiers when there is nothing interesting to say, especially when using the same naming scheme consistently.

    ``` motoko no-repl
    func map(x : Nat, y : Nat) : Nat { x + y };

    func eval(e : Expr) : Nat {
      let n =
        switch (e) {
          case (#neg(e1)) { - eval(e1) };
          case (#add(e1, e2)) { eval(e1) + eval(e2) };
          case (#mul(e1, e2)) { eval(e1) * eval(e2) };
        };
      Debug.print(n);
      return n;
    };
    ```

    Rationale: Contrary to popular belief, overly chatty local names can decrease readability instead of increasing it, by increasing the noise level.

-   In suitable cases, use plural form for describing a collection of items, such as a list or array.

    This also works for short names.

    ``` motoko no-repl
    func foreach<X>(xs : [X], f : X -> ()) {
      for (x in xs.vals()) { f(x) }
    }
    ```

Reply:
Yeah I actually read that but disagreed haha. But also, wiser people than me wrote that guide so I will concede